### PR TITLE
feature/ofv

### PR DESCRIPTION
**Pull request by david-lyder**
…JECTIVE FUNCTION VALUE WITHOUT CONSTANT"

### CODE REVIEW

File: parsers/nmparser/parse_lst_file.go [hunk 1]
```diff
@@ -5,28 +5,31 @@ import (
...
-func parseOFV(line string) float64 {
-	// get rid of everything but stars and ofv
-	result := line[10:]
-	result = strings.Replace(result, "*", "", -1)
-	output, _ := strconv.ParseFloat(strings.Fields(result)[0], 64)
-	return output
+func parseOFV(line string, ofvDetails *OfvDetails) {
+	if strings.Contains(line, "#OBJV") {
+		// get rid of everything but stars and ofv
+		result := line[10:]
```
Comment (dpastoor) [inline parsers/nmparser/parse_lst_file.go]: what's on the first 10 characters that makes it worth trimming off?

File: parsers/nmparser/parse_lst_file.go [hunk 2]
```diff
@@ -5,28 +5,31 @@ import (
...
+			strings.TrimSpace(strings.Replace(line, "OBJECTIVE FUNCTION VALUE WITH CONSTANT:", "", -1)),
+			64)
+	}
 }
 
 // ParseLstEstimationFile parses the lst file
 func ParseLstEstimationFile(lines []string) LstData {
-
-	var ofvIndex int
+	ofvDetails := new(OfvDetails)
```
Comment (dpastoor) [inline parsers/nmparser/parse_lst_file.go]: why `new` it vs just declaring the type `var ofvDeatils OfvDetails`

File: parsers/nmparser/parse_lst_file_test.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+func TestParTestParseOBJV2(t *testing.T) {
+	var lines = []string{
+		"OBJECTIVE FUNCTION VALUE WITHOUT CONSTANT: 821.705",
+		"OBJECTIVE FUNCTION VALUE WITH CONSTANT:       1639.561",
+		"N*LOG(2PI) CONSTANT TO OBJECTIVE FUNCTION:    817.855",
+		"#OBJV:********************************************    -7913.528       **************************************************",
+	}
+	lstData := ParseLstEstimationFile(lines)
+	if lstData.OFV.OFV != 817.855 || lstData.OFV.OFVWithConstant != 1639.561 || lstData.OFV.OFVNoConstant != -7913.528 {
```
Comment (dpastoor) [inline parsers/nmparser/parse_lst_file_test.go]: I'm not a fan of this `||` based testing structure. Namely, it does not scale well, especially if you have more complex structures to compare. In addition, it is unclear if you are testing the entire struct or just _parts_ of a struct. With testify you can just use `assert.Equal()` to compare against initialized structs with the correct value. 

```
expected := OfvDetails{
// correct values
}
assert.Equal(t, lstData.OFV, expected)
```

File: parsers/nmparser/parse_lst_file.go (hunk 1 above)
Comment (dpastoor) [reply parsers/nmparser/parse_lst_file.go]: aka probably worth noting as a comment something like,

line should look like #OBJV: *** <value> ***, though with many more stars. So they can see that you're trimming the #OBJV

Comment (david-lyder) [reply parsers/nmparser/parse_lst_file.go]: This was existing code that I should have re-visited.  I've normalized this code so it follows the same pattern as the others.

File: parsers/nmparser/parse_lst_file.go (hunk 2 above)
Comment (david-lyder) [reply parsers/nmparser/parse_lst_file.go]: It's a pointer. The pattern is to call the "parseOFV" function repeatedly while looping through the lines, each time possibly updating one of the 3 values in the OfvDetails structure.

File: parsers/nmparser/parse_lst_file_test.go (hunk 3 above)
Comment (david-lyder) [reply parsers/nmparser/parse_lst_file_test.go]: Yeah, assert is way better.

File: parsers/nmparser/parse_lst_file.go [hunk 4]
```diff
@@ -46,7 +48,13 @@ func ParseLstEstimationFile(lines []string) LstData {
 		case strings.Contains(line, "0DEFAULT SIGMA BOUNDARY"):
 			endParameterStucturesIndex = i
 		case strings.Contains(line, "#OBJV"):
-			ofvIndex = i
+			parseOFV(line, ofvDetails)
```
Comment (dpastoor) [inline parsers/nmparser/parse_lst_file.go]: could we actually change this to not take a pointer receiver, and instead use a struct directly and just return a modified version, to keep with the functional programming style of most of the parsing logic we have.

something like:

```
ofvDetails = parseOFV(line, ofvDetails)
```

**Pull request merged**